src/App.tsx

```diff
@@ -1,61 +1,126 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/Users';
 
 export const App = () => {
+  const [users] = useState<User[]>(usersFromServer);
+
+  const getUser = (userId: number) => {
+    const foundUser = usersFromServer.find(user => user.id === userId);
+
+    return foundUser || null;
+  };
+
+  const initialTodos: Todo[] = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUser(todo.userId),
+  }));
+
+  const [todos, setTodos] = useState(initialTodos);
+
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [selectFieldError, setSelectFieldError] = useState(false);
+
+  const validRegex = /^[A-Za-z\u0400-\u04FF0-9\s]+$/;
+
+  const inputValidation = (text: string) => {
+    return text.trim().match(validRegex);
+  };
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const getMaxId = (todoList: Todo[]) => {
+    const maxId = Math.max(...todoList.map(todo => todo.id));
+
+    return maxId + 1;
+  };
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    setTitleError(false);
+    setSelectFieldError(false);
+
+    if (!inputValidation(title) || title.trim() === '') {
+      setTitleError(true);
+
+      return;
+    }
+
+    if (!userId) {
```

Comment:
it should be possible to display 2 errors, see the working example:
![image](https://github.com/mate-academy/react_add-todo-form/assets/86923089/c0bac2a1-dd59-4b78-ad29-4421dcb66634)